### DISCUSSION TITLE
Is it consistent to name git branch `10.x` as `10.x-dev` instead of `dev-10.x`?

### DISCUSSION THREAD

**Discussion by bilogic** [composer/composer#11770] [Q&A / Support] [answered]
1. I refer to the well known naming of `dev-master`, `dev-main` and perhaps `dev-develop` when we require packages from a `path` locally
2. My package is on a git branch named `10.x`, so based on above, it seems more consistent for composer to refer to it as `dev-10.x`
3. However, what gets added to `composer.json` is always `10.x-dev`, what is the reason for this?
4. Is there a way to get `composer.json` to default to `dev-10.x` for consistency?

**Comment by Seldaek** [comment]
We distinguish between numeric branches (which get a -dev suffix to indicate they are a dev version) and non-numeric like main/.. which get a prefix, to clearly mark them as dev names, but also make sure they are non-sortable. The numeric ones are treated as numeric versions 10.x becoming 10.99999 internally, but the non numeric ones just are requirable by their name. IMO it is clearer to have the different naming conventions rather than the appearance of consistency as the way they are handled is very much inconsistent.

**Reply by bilogic** [reply]
Ok, so if git branch starts with numeric, `dev` is appended as `-dev`
If it starts with non-numeric, `dev` is prepended as `dev-`

Correct?

**Reply by Seldaek** [reply]
That is correct :)

**Reply by Seldaek** [reply, marked as answer]
Well sorry, it's a bit more strict than that for it to be considered numeric it has to match here https://github.com/composer/semver/blob/main/src/VersionParser.php#L216-L216 - not just start with a number.

**Reply by bilogic** [reply]
Well, as long as the logic is codified. Thank you!
